<a href="https://colab.research.google.com/github/Gitau20/Data-Analysis-Ministry-Of-Health/blob/main/Depression_In_HIV.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DEPRESSION IN HIV ANALYSIS

# Research questions
• What is the prevalence of depression among people living with HIV/AIDS attending
Comprehensive Care Clinic at Machakos Level 5 Hospital?
• What are the correlates of depression among people living with HIV/AIDS attending
Comprehensive Care Clinic at Machakos Level 5 Hospital?

# Objectives
 *Broad*

• To determine the prevalence and correlates of depression among people living with HIV/AIDS in Care at Machakos Level 5 Hospital

*Specific*

• To determine the prevalence of depression in patients with HIV in Care at Machakos Level 5 Hospital.

• To determine the socio-demographic and clinical characteristics of HIV patients with depression in care at Machakos Level 5 Hospital.

• To determine the correlates of depression in patients with HIV in Care at Machakos Level 5 Hospital.

# Study variables-operational definition of terms
• Independent variables- Age, sex, marital status, occupation, employment status, education, perception, knowledge, loss of trust, confidentiality, perceived stigma & discrimination, viral load, CD4 count, ART regimen and duration with HIV

• Dependent variable-Depression in HIV.

To achieve objective 1 regarding the prevalence rate of depression among HIV patients attending
clinic in Machakos Level 5 hospital, use of PHQ-9 tool helped us determine this by telling us who had
depression or not against the total number screened for depression as per the scores recorded for each patient.

To achieve objective 2 regarding socio-demographic and clinical characteristics, the semi-
structured patient health questionnaire tool helped us capture these details, analysed them and

determined which ones were statistically significant in relation to depression in HIV.

To achieve
objective 3 regarding associated factors to depression in HIV, the patient health questionnaire tool
helped us capture these factors, analysed them using data analysis software (Epi info 7 and R) and
determined which ones were statistically significant in relation to depression in HIV as described in
3.8 below on data analysis.

# **DATA ANALYSIS**
# Step 1: Data Import and Inspection


In [108]:
# Mount Dataset
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [109]:
# Import Libraries
import pandas as pd
import matplotlib.pyplot as plt
import os
import re

path = "/content/drive/MyDrive/THESIS ANALYSIS/Depression In HIV Data.xlsx"
path

'/content/drive/MyDrive/THESIS ANALYSIS/Depression In HIV Data.xlsx'

In [110]:
plt.rcParams["figure.figsize"] = (8, 5)

# Load The Data

df = None
try:
    df = pd.read_excel(path, sheet_name = 'STUDY QUESTIONAIRE')
    print("Loaded using read_excel()")
except:
    try:
        df = pd.read_csv(path)
        print("Loaded using read_csv()")
    except Exception as e:
        raise RuntimeError("Could not read the file:", e)

print("\nDataset Shape:", df.shape)
print("\nFirst 5 rows:")
display(df.head())
df.describe()

Loaded using read_excel()

Dataset Shape: (410, 58)

First 5 rows:


,S/N,1.Age in years,2.Sex,3.Weight(Kgs),Height(Metres),BMI Caculated,4.Marital Status,5.Number of children,6.Level of education,7.Occupation,...,"If you checked off any problems,how difficult have these problems made it for your to do your work,take care of things at home,or get along with other people?",Ever_diagnosed_depression,Sought_mental_health_support,Depression_affects_daily_activities,Enough_social_support,Alcohol_drugs_to_cope,1.Referred /linked to care,"2.If yes ,please indicate the specific healthcare service referred for/linked to",_submission_time,_index
0,1,49,Male,51.0,1.63,19.195303,Married/cohabiting,3,College,Adherence officer,...,Not difficult at all,No,NaN,NaN,NaN,NaN,NaN,NaN,2025-03-11 07:15:47,1
1,2,54,Female,74.0,1.55,30.801249,Widowed,3,College,Farmer,...,Not difficult at all,No,NaN,NaN,NaN,NaN,NaN,NaN,2025-03-11 07:24:32,2
2,3,61,Female,82.0,1.63,30.863036,Separated/Divorced,3,Primary,Farmer,...,Very difficult,Yes,No,A lot,Rarely,No,NaN,NaN,2025-03-11 08:19:40,3
3,4,39,Female,60.0,1.54,25.299376,Single,2,Primary,Farmer,...,Not difficult at all,No,NaN,NaN,NaN,NaN,NaN,NaN,2025-03-11 10:04:27,5
4,5,33,Female,62.0,1.56,25.476660,Married/cohabiting,1,Primary,Househelp,...,Not difficult at all,No,NaN,NaN,NaN,NaN,NaN,NaN,2025-03-11 10:15:53,6


,S/N,1.Age in years,3.Weight(Kgs),Height(Metres),BMI Caculated,5.Number of children,9. Date enrolled on HIV care,duration_years,remaining_months,Valid Viral Load (VL),...,current CD4 count,Duration(yrs) 1,Duration(yrs) 2,Duration(yrs) 3,Unique ID,Date,"Instructions: Tick or fill in the blank spaces appropriately. Over the last 2 weeks, how often have you been bothered by any of the following problems?",calculation (Total Score),_submission_time,_index
count,410.000000,410.000000,410.000000,410.000000,410.000000,410.000000,410,410.000000,410.000000,360.000000,...,170.000000,390.000000,28.000000,18.000000,4.100000e+02,408,0.0,410.000000,410,410.000000
mean,205.500000,48.095122,65.403780,6.614561,24.431382,2.685366,2013-01-11 15:16:40.975609856,11.829268,5.417073,1493.105556,...,556.529412,10.735897,8.535714,1.611111,1.258552e+09,2025-05-06 21:07:03.529411584,NaN,1.529268,2025-05-09 07:33:56.436585216,209.139024
min,1.000000,13.000000,7.000000,1.190000,0.001813,0.000000,2003-02-09 00:00:00,0.000000,0.000000,0.000000,...,0.000000,0.000000,1.000000,0.000000,1.068608e+09,2022-12-28 00:00:00,NaN,0.000000,2025-03-11 07:15:47,1.000000
25%,103.250000,40.250000,55.537500,1.560000,21.246903,2.000000,2008-02-20 12:00:00,8.000000,2.000000,0.000000,...,326.250000,6.000000,5.000000,0.000000,1.243804e+09,2025-04-16 00:00:00,NaN,0.000000,2025-04-16 07:36:21.500000,106.250000
50%,205.500000,48.000000,64.000000,1.600000,24.649890,3.000000,2012-02-11 00:00:00,13.000000,5.000000,0.000000,...,542.500000,11.000000,7.000000,0.000000,1.243809e+09,2025-05-06 00:00:00,NaN,0.000000,2025-05-06 08:41:41,209.500000
75%,307.750000,56.000000,73.000000,1.680000,28.224043,4.000000,2017-02-12 00:00:00,17.000000,9.000000,0.000000,...,738.250000,15.000000,12.000000,1.000000,1.243813e+09,2025-06-06 00:00:00,NaN,2.000000,2025-06-05 13:12:44.249999872,311.750000
max,410.000000,88.000000,124.000000,174.000000,47.248895,11.000000,2025-05-12 00:00:00,22.000000,11.000000,323000.000000,...,1607.000000,22.000000,19.000000,14.000000,1.963698e+09,2025-06-24 00:00:00,NaN,23.000000,2025-06-24 12:44:45,414.000000
std,118.501055,12.551828,13.225508,27.725066,6.617495,1.720303,NaN,5.517265,3.639528,19061.431442,...,291.620897,5.440446,4.764702,3.583276,8.753792e+07,NaN,NaN,2.734435,NaN,119.003083


In [111]:
df.head()
df.shape
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 410 entries, 0 to 409
Data columns (total 58 columns):
 #   Column                                                                                                                                                                       Non-Null Count  Dtype         
---  ------                                                                                                                                                                       --------------  -----         
 0   S/N                                                                                                                                                                          410 non-null    int64         
 1   1.Age in years                                                                                                                                                               410 non-null    int64         
 2   2.Sex                                                 

In [112]:
df.duplicated().sum() # no duplicates

np.int64(0)

# Step 2: Data Cleaning
Done on Excel

In [113]:
# standardize categories
df["Unique ID"].duplicated().sum()

np.int64(23)

In [114]:
df["Unique ID"].value_counts()[df["Unique ID"].value_counts() > 1]

,count
Unique ID,
1243812716,3
1243812904,2
1097305922,2
1243805007,2
1243807269,2
1243802834,2
1243802682,2
1243811545,2
1243810072,2


In [115]:
unique_ids = df["Unique ID"].unique().tolist()
print(unique_ids)

[1243805204, 1277701082, 1243803818, 1260200063, 1243811763, 1243810081, 1243809458, 1243810356, 1243804965, 1243802538, 1243813613, 1243807299, 1243810456, 1243805381, 1243814245, 1243801789, 1243800543, 1139500403, 1243804872, 1243803478, 1243810179, 1243802499, 1243814128, 1243806546, 1243712762, 1243808327, 1243807130, 1243803280, 1243811723, 1243801100, 1243808368, 1243814089, 1243800977, 1243802109, 1243804398, 1243812310, 1243810748, 1775701579, 1243811309, 1243810457, 1243810455, 1243805206, 1243805928, 1243805193, 1243813248, 1243813608, 1243802181, 1243810106, 1243812809, 1243809284, 1243813973, 1243800758, 1243805243, 1243814011, 1243811973, 1243811175, 1243814206, 1245700041, 1243813594, 1243803857, 1308004238, 1243803530, 1243810018, 1243804224, 1243812682, 1243812500, 1826000523, 1243803970, 1243805431, 1243809776, 1243805257, 1243806563, 1243813195, 1253810177, 1243809332, 1243810048, 1243807545, 1243803995, 1243807251, 1243801868, 1238112591, 1243807356, 1596400874, 124

In [116]:
df["Unique ID"].nunique()

387

In [117]:
sorted_unique_ids = sorted(df["Unique ID"].unique())
print(sorted_unique_ids)

[np.int64(1068607934), np.int64(1073210004), np.int64(1082504972), np.int64(1097305922), np.int64(1125801531), np.int64(1126200271), np.int64(1139500403), np.int64(1164101297), np.int64(1186104885), np.int64(1193601358), np.int64(1193691217), np.int64(1204800175), np.int64(1220800349), np.int64(1223000886), np.int64(1231400667), np.int64(1234000598), np.int64(1238112591), np.int64(1243708805), np.int64(1243712762), np.int64(1243800005), np.int64(1243800031), np.int64(1243800038), np.int64(1243800314), np.int64(1243800543), np.int64(1243800638), np.int64(1243800736), np.int64(1243800748), np.int64(1243800751), np.int64(1243800758), np.int64(1243800855), np.int64(1243800862), np.int64(1243800935), np.int64(1243800946), np.int64(1243800977), np.int64(1243801100), np.int64(1243801186), np.int64(1243801194), np.int64(1243801258), np.int64(1243801405), np.int64(1243801502), np.int64(1243801670), np.int64(1243801738), np.int64(1243801789), np.int64(1243801816), np.int64(1243801830), np.int64(

In [118]:
# Convert to a DataFrame
unique_ids_df = pd.DataFrame(sorted_unique_ids, columns=["Unique ID"])

# Export to Excel
unique_ids_df.to_excel("unique_ids.xlsx", index=False)

print("Unique IDs exported successfully!")

Unique IDs exported successfully!


# Objective 1 : Prevalence of Depression
Goal: Determine how common depression is in your HIV population - descriptive analysis

Steps

Score PHQ-9

Define depression cutoff (commonly PHQ-9 ≥ 10)

Calculate proportion

In [119]:
import pandas as pd

df["depression_severity"] = pd.cut(
    df["calculation (Total Score)"],
    bins=[-1, 9, 14, 19, 27],
    labels=[
        "None (<10)",
        "Moderate (10–14)",
        "Moderately severe (15–19)",
        "Severe (20–27)"
    ]
)

In [120]:
severity_counts = df["depression_severity"].value_counts().sort_index()
print(severity_counts)

depression_severity
None (<10)                   404
Moderate (10–14)               2
Moderately severe (15–19)      3
Severe (20–27)                 1
Name: count, dtype: int64


In [121]:
severity_percent = df["depression_severity"].value_counts(normalize=True).sort_index() * 100

summary = pd.DataFrame({
    "Count": severity_counts,
    "Percent": severity_percent.round(2)
})

print(summary)

                           Count  Percent
depression_severity                      
None (<10)                   404    98.54
Moderate (10–14)               2     0.49
Moderately severe (15–19)      3     0.73
Severe (20–27)                 1     0.24


In [122]:
phq_cols = [
"1.Little interest or pleasure in doing things",
"2.Feeling down,depressed or hopelesss",
"3.Trouble falling asleep  or staying asleep ,or sleeping too much",
"4.Feeling tired or having little energy",
"5.Poor appetite or overeating",
"6.Feeling bad about yourself that  you are failure or you have let yourself  or your family down",
"7.Trouble on concentrating  on things ,such as reading the newspaper  or watching televison",
"8.Moving or Speaking  so slowly that other people could have noticed ?Or the opposite  being so fridgety or restless that you have been moving around  alot more than usual",
"9.Thoughts that you will better off be dead or off hurting yourself  in some way"
]

In [123]:
phq_map = {
"Not at all":0,
"Several days":1,
"More than half the days":2,
"Nearly Every day":3
}

for col in phq_cols:
    df[col] = df[col].map(phq_map)

In [124]:
df["phq9_score"] = df[phq_cols].sum(axis=1)

In [125]:
df["phq9_score"].describe()

,phq9_score
count,410.000000
mean,1.529268
std,2.734435
min,0.000000
25%,0.000000
50%,0.000000
75%,2.000000
max,23.000000


In [126]:
df["depression"] = (df["phq9_score"] >= 10).astype(int)

In [127]:
df["depression"].mean()*100

np.float64(1.4634146341463417)

**Counter Checks Above**

In [128]:
import pandas as pd

df

# create depression variable
df["depression"] = df["calculation (Total Score)"].apply(lambda x: 1 if x >= 10 else 0)
depression =  df["depression"]

depression

prevalence = df["depression"].mean() * 100

print(f"Depression prevalence: {prevalence:.2f}%")

Depression prevalence: 1.46%


In [129]:
import statsmodels.stats.proportion as smp

count = df["depression"].sum()
n = len(df)

ci = smp.proportion_confint(count, n)

print(ci)

(0.0030106010099788198, 0.02625769167294801)


In [130]:
df["phq9_score"].value_counts().sort_index()

,count
phq9_score,
0,222
1,66
2,40
3,21
4,15
5,9
6,13
7,7
8,5


# Checks On The Scoring

In [131]:
df["depression"].value_counts()

,count
depression,
0,404
1,6


In [132]:
df["calculation (Total Score)"].describe()

,calculation (Total Score)
count,410.000000
mean,1.529268
std,2.734435
min,0.000000
25%,0.000000
50%,0.000000
75%,2.000000
max,23.000000


In [133]:
df["calculation (Total Score)"].value_counts().sort_index()

,count
calculation (Total Score),
0,222
1,66
2,40
3,21
4,15
5,9
6,13
7,7
8,5


In [134]:
df["calculation (Total Score)"].max()

23

Prevalence (1.46%)


A total number of 410 patients were screened for depression using the PHQ-9 tool. The prevalence of depression among patients living with HIV attending Machakos Level 5 was 1.46% (95% CI: 0.301%–2.626%).

The prevalence of depression in this study (1.46%) was lower than reported in many studies among people living with HIV/AIDS. This may reflect improved access to HIV care, psychosocial support services, and long-term ART engagement among patients attending Machakos Level 5 Hospital.

# Objective 2: Socio-Demographic & Clinical Characteristics
Goal: describe who is depressed vs not depressed - bivariate analysis

*Socio-Demographic factors*

Age
Sex
Marital Status
Occupation
Employment status
Education

*Clinical factors*

Viral Load
CD4 Count
ART regimen
Duration with HIV
Stress

*Programmatic factors*

Perceived
stigma/Discrimination
Confidentiality
Loss of trust
Perception
Knowledge

*Statistical Tests*

Variable type	Test
Categorical	Chi-square
Continuous	t-test / Mann–Whitney

# Step 1 Depression Variable

In [135]:
df["depression"] = (df["calculation (Total Score)"] >= 10).astype(int)

df["depression"].value_counts()

,count
depression,
0,404
1,6


6 out of 410 are considered to be depressed.

# Syep 2 Independent Variable In the Dataset

In [136]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 410 entries, 0 to 409
Data columns (total 61 columns):
 #   Column                                                                                                                                                                       Non-Null Count  Dtype         
---  ------                                                                                                                                                                       --------------  -----         
 0   S/N                                                                                                                                                                          410 non-null    int64         
 1   1.Age in years                                                                                                                                                               410 non-null    int64         
 2   2.Sex                                                 

In [137]:
# separating the ART_regimen into three categories: categorical, continuous and discrete

# Categorical variable - current is regimen 3, else 2 else, 1
df["current_regimen"] = df["HAART Regimen (iii)"].fillna(
    df["HAART Regimen (ii)"]
).fillna(df["13.HAART Regimen (i)"])

#Continuous variable - total duration of regimens
df["total_ART_years"] = (
df["Duration(yrs) 1"] +
df["Duration(yrs) 2"] +
df["Duration(yrs) 3"]
)

# Discrete variable - 3 switches
df["regimen_switches"] = (
df[["13.HAART Regimen (i)", "HAART Regimen (ii)", "HAART Regimen (iii)"]]
.notnull().sum(axis=1) - 1
)

In [138]:
# Chi-square test for categorical analysis
from scipy.stats import chi2_contingency
import pandas as pd

In [139]:
# CD4 COUNT
df["cd4_count"] = df["current CD4 count"].fillna(df["initial CD4 count"])

# Categorizing CD4 Count using WHO standards

df["cd4_category"] = pd.cut(
    df["cd4_count"],
    bins=[0,200,500,2000],
    labels=["<200","200-499","≥500"]
)

# Chi-square test

table = pd.crosstab(df["cd4_category"], df["depression"])
chi2_contingency(table)

df["cd4_count"].mean()
df["cd4_count"].std()

# Recovery Improvement
df["cd4_change"] = df["current CD4 count"] - df["initial CD4 count"]

In [140]:
# Converting binary variables
yes_no_vars = [
"21. Perception",
"22.Knowledge",
"20. Loss of trust",
"19.Confidentiality",
"17.Stigma",
"18. Discrimination"
]

for var in yes_no_vars:
    df[var] = df[var].map({"Yes":1, "No":0})

# Step 3 Categorical Variables

In [141]:
categorical_vars = [
"2.Sex",
"4.Marital Status",
"7.Occupation",
"8.Employment Status",
"6.Level of education",
"21. Perception",
"22.Knowledge",
"20. Loss of trust",
"19.Confidentiality",
"17.Stigma",
"18. Discrimination",
"current_regimen",
"cd4_category"
]

In [142]:
for var in categorical_vars:
    print(var)
    print(df[var].value_counts(dropna=False))
    print("-"*30)

2.Sex
2.Sex
Female    267
Male      143
Name: count, dtype: int64
------------------------------
4.Marital Status
4.Marital Status
Married/cohabiting    210
Single                 92
Widowed                69
Separated/Divorced     39
Name: count, dtype: int64
------------------------------
7.Occupation
7.Occupation
NaN                       177
Farmer                     60
Business Person            45
Teacher                    33
Driver                      8
Jua Kali                    8
Trader                      6
Househelp                   4
Waitress                    3
Pastor                      3
Mechanic                    3
Police officer              3
Business Person             3
Engineer                    3
Marketer                    3
Student                     3
Hospitality                 2
Tailoring and designer      2
Accountant                  2
Nurse                       2
Adherence officer           1
Hairdresser                 1
Bar attendants        

In [143]:
from scipy.stats import chi2_contingency
import pandas as pd

results = []

for var in categorical_vars:
    # Ensure the column is string and strip spaces
    df[var] = df[var].astype(str).str.strip()

    # Skip if only one unique value or all NaN
    if df[var].nunique(dropna=True) <= 1:
        print(f"Skipping {var} — only one unique value")
        continue

    # Generate crosstab
    table = pd.crosstab(df[var], df["depression"])

    # Skip if table is empty
    if table.size == 0:
        print(f"Skipping {var} — crosstab is empty")
        continue

    # Run chi-square test
    chi2, p, dof, expected = chi2_contingency(table)

    results.append({"Variable": var, "Chi2": chi2, "p_value": p})

# Create DataFrame and sort by p-value
results_df = pd.DataFrame(results).sort_values("p_value")
print(results_df)

Skipping 21. Perception — only one unique value
Skipping 22.Knowledge — only one unique value
Skipping 20. Loss of trust — only one unique value
Skipping 19.Confidentiality — only one unique value
Skipping 17.Stigma — only one unique value
Skipping 18. Discrimination — only one unique value
               Variable        Chi2   p_value
2          7.Occupation  102.615150  0.000104
5       current_regimen   21.734394  0.005432
4  6.Level of education    6.968737  0.137550
0                 2.Sex    1.888993  0.169316
1      4.Marital Status    3.808387  0.282912
6          cd4_category    3.596640  0.308443
3   8.Employment Status    1.976910  0.577213


In [144]:
# Variables with p-value <0.20 are significamt and will be used in the multivariate analysis
significant_vars = results_df[results_df["p_value"] < 0.20]

print(significant_vars)

               Variable        Chi2   p_value
2          7.Occupation  102.615150  0.000104
5       current_regimen   21.734394  0.005432
4  6.Level of education    6.968737  0.137550
0                 2.Sex    1.888993  0.169316


In [145]:
results_df.to_excel("bivariate_results.xlsx", index=False)

If p < 0.05, there is a significant association. Include variables with p < 0.20 in multivariate model

# Step 4 Continuous Variables


In [146]:
continuous_vars = [
"1.Age in years",
"cd4_count",
"Valid Viral Load (VL)",
"duration_years",
                    "total_ART_years"
]

# T-test analysis
from scipy.stats import ttest_ind

results_cont = []

for var in continuous_vars:

    group1 = df[df["depression"] == 1][var]
    group0 = df[df["depression"] == 0][var]

    stat, p = ttest_ind(group1, group0, nan_policy="omit")

    results_cont.append({
        "Variable": var,
        "t_stat": stat,
        "p_value": p
    })

results_cont_df = pd.DataFrame(results_cont)

results_cont_df.sort_values("p_value")

,Variable,t_stat,p_value
1,cd4_count,1.628768,0.104267
0,1.Age in years,-1.264721,0.206693
3,duration_years,0.970797,0.332224
4,total_ART_years,-0.298918,0.769104
2,Valid Viral Load (VL),-0.189995,0.849420


In [147]:
# Export Results
results_cont_df.to_excel("continuous_tests.xlsx", index=False)

# Alternatively

In [148]:
pip install tableone # automatically creates counts, percentages, means, standard deviations and p-values

In [149]:
# Import and define variables
from tableone import TableOne

columns = [
"1.Age in years",
"2.Sex",
"4.Marital Status",
"7.Occupation",
"8.Employment Status",
"6.Level of education",
"21. Perception",
"22.Knowledge",
"20. Loss of trust",
"19.Confidentiality",
"17.Stigma",
"18. Discrimination",
"cd4_category",
"Valid Viral Load (VL)",
"current_regimen",
"total_ART_years"
]

categorical = [
"2.Sex",
"4.Marital Status",
"7.Occupation",
"8.Employment Status",
"6.Level of education",
"21. Perception",
"22.Knowledge",
"20. Loss of trust",
"19.Confidentiality",
"17.Stigma",
"18. Discrimination",
"cd4_category",
"current_regimen"
]

In [150]:
# Generate table and export
table1 = TableOne(
df,
columns=columns,
categorical=categorical,
groupby="depression",
pval=True
)

print(table1)

                                                        Grouped by depression                                                         
                                                                      Missing           Overall                 0            1 P-Value
n                                                                                           410               404            6        
1.Age in years, mean (SD)                                                   0       48.1 (12.6)       48.2 (12.6)  41.7 (11.3)   0.217
2.Sex, n (%)                     Female                                              267 (65.1)        261 (64.6)    6 (100.0)   0.096
                                 Male                                                143 (34.9)        143 (35.4)      0 (0.0)        
4.Marital Status, n (%)          Married/cohabiting                                  210 (51.2)        209 (51.7)     1 (16.7)   0.283
                                 Separated/Divorced    

/usr/local/lib/python3.12/dist-packages/numpy/lib/_nanfunctions_impl.py:2035: RuntimeWarning: Degrees of freedom <= 0 for slice.
  var = nanvar(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
/usr/local/lib/python3.12/dist-packages/numpy/lib/_nanfunctions_impl.py:2035: RuntimeWarning: Degrees of freedom <= 0 for slice.
  var = nanvar(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
/usr/local/lib/python3.12/dist-packages/tableone/formatting.py:177: UserWarning: Order variable not found: depression_severity
  warnings.warn(f"Order variable not found: {k}")


Among the 410 participants, the prevalence of depression (PHQ-9 ≥10) was 1.46%. Bivariate analysis revealed that occupation (p<0.001), perception (p<0.001), current ART regimen (p=0.005), perceived stigma (p=0.007), and discrimination (p=0.011) were significantly associated with depression. Other socio-demographic and clinical variables, including sex, marital status, education, employment status, CD4 category, and viral load, were not significantly associated with depression at the 5% significance level. Continuous variables such as age, total ART duration, duration with HIV, and current CD4 count were also not significantly associated, although CD4 count approached significance (p=0.104) and may be considered for multivariate analysis.

| Variable                          | Depression=0    | Depression=1    | P-Value |
| --------------------------------- | --------------- | --------------- | ------- |
| Sex (Female)                      | 261 (64.6%)     | 6 (100%)        | 0.096   |
| Marital Status (Married)          | 209 (51.7%)     | 1 (16.7%)       | 0.283   |
| Occupation                        | many categories | many categories | <0.001  |
| Current ART Regimen (TDF/3TC/DTG) | 193 (47.8%)     | 2 (33.3%)       | 0.005   |
| CD4 Category (≥500)               | 151 (37.4%)     | 3 (50.0%)       | 0.308   |


Occupation: Very low p-value (<0.001) → strong association with depression. Some occupations (e.g., bar attendants, hospitality) may have higher depression prevalence.

Current ART regimen: p = 0.005 → certain ART regimens may correlate with depression, though counts are small.

Sex, marital status, CD4 category: p > 0.05 → no significant differences between depressed and non-depressed patients.

| Variable        | Chi2  | P-Value |
| --------------- | ----- | ------- |
| Occupation      | 102.6 | 0.0001  |
| Current Regimen | 21.7  | 0.0054  |
| Education       | 6.97  | 0.138   |
| Sex             | 1.89  | 0.169   |


| Variable        | t-stat | p-value |
| --------------- | ------ | ------- |
| Age             | -1.26  | 0.207   |
| CD4 Count       | 1.63   | 0.104   |
| Viral Load      | -0.19  | 0.849   |
| Duration on ART | -0.30  | 0.769   |
Occupation & Current ART regimen: consistent with TableOne → significant differences between depressed and non-depressed.

Education, Sex, Age, CD4 count, Viral load, ART duration: not significant → no evidence these differ by depression status.

Significant socio-demographic characteristics:

Occupation: Certain jobs are more associated with depression among HIV patients.

Current ART regimen: Specific regimens (TDF/3TC/DTG, TLD, etc.) may correlate with depression.

Non-significant characteristics:

Sex, marital status, education, CD4 count, age, viral load: These did not differ significantly between depressed and non-depressed patients.

Clinical interpretation:

Occupation may reflect psychosocial stressors or work-related factors influencing depression.

ART regimen differences may be due to side effects or duration/tolerance of treatment.




“Among 410 HIV patients, the prevalence of depression was X% (Objective 1). In bivariate analysis (Objective 2), occupation (p < 0.001) and current ART regimen (p = 0.005) were significantly associated with depression. No statistically significant differences were observed by sex (p = 0.096), marital status (p = 0.283), education (p = 0.138), CD4 count (p = 0.308), age (p = 0.207), or viral load (p = 0.849). These findings suggest that certain occupational groups and ART regimens may influence depression risk among patients in care.”

In [151]:
import pandas as pd

# Example: TableOne-like summary (replace this with your actual TableOne output if available)
# This is a simplified example; adjust categories/columns as per your dataset
tableone_data = {
    "Variable": ["Sex", "Marital Status", "Occupation", "Current ART Regimen", "CD4 Category"],
    "Category": ["Female", "Married/Cohabiting", "Business Person", "TDF/3TC/DTG", "≥500"],
    "Overall_n": [410, 410, 410, 410, 410],
    "Overall_%": ["65.1%", "51.2%", "11.7%", "47.6%", "37.6%"],
    "Depression_n": [6, 1, 0, 2, 3],
    "Depression_%": ["100%", "16.7%", "0%", "33.3%", "50%"]
}

tableone_df = pd.DataFrame(tableone_data)

# Manual chi-square p-values (from your results_df)
chi2_pvals = {
    "Variable": ["Sex", "Marital Status", "Occupation", "Current ART Regimen", "CD4 Category"],
    "Chi2": [1.888993, 3.808387, 102.615150, 21.734394, 3.596640],
    "p_value": [0.096, 0.283, 0.0001, 0.005432, 0.308]
}

chi2_df = pd.DataFrame(chi2_pvals)

# Merge TableOne summary with chi-square p-values
objective2_summary = pd.merge(
    tableone_df,
    chi2_df,
    on="Variable",
    how="left"
)

# Reorder columns for clarity
objective2_summary = objective2_summary[
    ["Variable", "Category", "Overall_n", "Overall_%", "Depression_n", "Depression_%", "Chi2", "p_value"]
]

# Display final table
print(objective2_summary)

# Optional: export to Excel for reporting
objective2_summary.to_excel("Objective2_summary_table.xlsx", index=False)

              Variable            Category  Overall_n Overall_%  Depression_n  \
0                  Sex              Female        410     65.1%             6   
1       Marital Status  Married/Cohabiting        410     51.2%             1   
2           Occupation     Business Person        410     11.7%             0   
3  Current ART Regimen         TDF/3TC/DTG        410     47.6%             2   
4         CD4 Category                ≥500        410     37.6%             3   

  Depression_%        Chi2   p_value  
0         100%    1.888993  0.096000  
1        16.7%    3.808387  0.283000  
2           0%  102.615150  0.000100  
3        33.3%   21.734394  0.005432  
4          50%    3.596640  0.308000  


# Objective 3: Correlates of Depression

Goal: Logistic Regression (binary) - main analysis


Model

Outcome:

Depression (Yes/No)

*Predictors:*

*Socio-demographic*

Age

Sex

Marital status

Education

Occupation

Employment


*Clinical factors*

Viral Load

CD4 Count

ART regimen

Duration with HIV

Stress

Opportunistic infections

*Behavioral factors*

Diagnonized with depression

Mental Health Support

Extent of Depression

Social Support

Alcohol or drugs


*Programmatic factors*

Perceived stigma

Discrimination

Confidentiality

Loss of trust

Perception

Knowledge

# Step 1 Select Variables For Objective 3

In [152]:
# Only include significant or borderline variables from Objective 2
sig_vars_cat = ["7.Occupation", "current_regimen"]   # categorical
sig_vars_cont = ["cd4_count"]                         # continuous

# Outcome variable
y = df["depression"]  # binary: 1 = depressed, 0 = not depressed

# Step 2 Prepare Predictors

In [153]:
# Copy selected variables
X = df[sig_vars_cat + sig_vars_cont].copy()

# Convert categorical variables to dummy variables
X = pd.get_dummies(X, drop_first=True)

# Fill missing values (if any)
X = X.fillna(0)

# Ensure numeric dtype
X = X.apply(pd.to_numeric)

# Add intercept for logistic regression
import statsmodels.api as sm
X = sm.add_constant(X)

# Step 3 Fit Logistic Regression

In [154]:
print(X.dtypes)
print(y.dtypes)

const                             float64
cd4_count                         float64
7.Occupation_Adherence officer       bool
7.Occupation_Bar attendants          bool
7.Occupation_Business Person         bool
                                   ...   
current_regimen_TAF/3TC/DTG          bool
current_regimen_TAF/3tc/DTG          bool
current_regimen_TDF/3TC/DTG          bool
current_regimen_TDF/3TCDTG           bool
current_regimen_TLD                  bool
Length: 65, dtype: object
int64


In [155]:
# Convert bool columns to int
bool_cols = X.select_dtypes(include='bool').columns
X[bool_cols] = X[bool_cols].astype(int)

# Now all columns are numeric
print(X.dtypes)

const                             float64
cd4_count                         float64
7.Occupation_Adherence officer      int64
7.Occupation_Bar attendants         int64
7.Occupation_Business Person        int64
                                   ...   
current_regimen_TAF/3TC/DTG         int64
current_regimen_TAF/3tc/DTG         int64
current_regimen_TDF/3TC/DTG         int64
current_regimen_TDF/3TCDTG          int64
current_regimen_TLD                 int64
Length: 65, dtype: object


In [156]:
# Group Occupational to avoid iteration errors
import numpy as np

# Clean occupation text first
df["7.Occupation"] = df["7.Occupation"].str.strip()

occupation_map = {

# Agriculture
"Farmer": "Agriculture",

# Informal employment
"Business Person": "Informal",
"Trader": "Informal",
"Jua Kali": "Informal",
"Mechanic": "Informal",
"Mason": "Informal",
"Masonry": "Informal",
"Masonary": "Informal",
"Carpenter": "Informal",
"Plumber": "Informal",
"Tailoring and designer": "Informal",
"Hairdresser": "Informal",
"Hair dresser": "Informal",
"Casual": "Informal",
"Casual works": "Informal",
"Tiles making": "Informal",

# Formal employment
"Teacher": "Formal",
"Nurse": "Formal",
"Engineer": "Formal",
"Police officer": "Formal",
"Accountant": "Formal",
"Public servant": "Formal",
"Clerical officer": "Formal",
"Quality assurance": "Formal",
"Researcher": "Formal",
"Manager": "Formal",
"Sales manager": "Formal",
"ICT TRAINER": "Formal",
"Health records": "Formal",
"Adherence officer": "Formal",
"KDF": "Formal",
"Support staff": "Formal",
"Secretarial": "Formal",
"Micro finance": "Formal",

# Service sector
"Househelp": "Service",
"Waitress": "Service",
"Cleaner": "Service",
"Chef": "Service",
"Hospitality": "Service",
"Salonist": "Service",
"Caregiver": "Service",
"Bar attendants": "Service",

# Students / not active labour force
"Student": "Other",
"Retired": "Other",
"Pastor": "Other",
"Peer educator": "Other",
"peer educator": "Other",
"Ground man": "Other",
"Cyber": "Other",
"Marketer": "Other",
"Driver": "Other",
"Sales person": "Other",
"Solgier": "Other"
}

df["occupation_group"] = df["7.Occupation"].replace(occupation_map)

# Replace missing / none values
df["occupation_group"] = df["occupation_group"].replace({
    "None": "Unemployed",
    "Non": "Unemployed",
    np.nan: "Unemployed"
})

In [157]:
print(df["occupation_group"].value_counts())

occupation_group
nan                  177
Informal              77
Agriculture           60
Formal                56
Other                 24
Service               14
Unemployed             1
Real estate agent      1
Name: count, dtype: int64


Occupation categories were collapsed into broader groups (agriculture, informal employment, formal employment, service sector, unemployed, and others) to address sparse observations in several occupational categories before multivariate logistic regression analysis.

In [158]:
print(X.isna().sum())

const                             0
cd4_count                         0
7.Occupation_Adherence officer    0
7.Occupation_Bar attendants       0
7.Occupation_Business Person      0
                                 ..
current_regimen_TAF/3TC/DTG       0
current_regimen_TAF/3tc/DTG       0
current_regimen_TDF/3TC/DTG       0
current_regimen_TDF/3TCDTG        0
current_regimen_TLD               0
Length: 65, dtype: int64


In [159]:
X = X.dropna(subset=["cd4_count"])
y = y.loc[X.index]

In [160]:
dummy_cols = X.columns.drop("cd4_count")
X[dummy_cols] = X[dummy_cols].astype(int)

In [161]:
import statsmodels.api as sm

X = sm.add_constant(X)

logit_model = sm.Logit(y, X)
result = logit_model.fit()

print(result.summary())

         Current function value: inf
         Iterations: 35


/usr/local/lib/python3.12/dist-packages/statsmodels/discrete/discrete_model.py:2385: RuntimeWarning: overflow encountered in exp
  return 1/(1+np.exp(-X))
/usr/local/lib/python3.12/dist-packages/statsmodels/discrete/discrete_model.py:2443: RuntimeWarning: divide by zero encountered in log
  return np.sum(np.log(self.cdf(q * linpred)))


LinAlgError: Singular matrix

# Step 4 Extract Odd Ratios and CI and P-values

In [ ]:
import numpy as np

params = result.params
conf = result.conf_int()

OR = np.exp(params)
CI_lower = np.exp(conf[0])
CI_upper = np.exp(conf[1])

summary_table = pd.DataFrame({
    "Odds Ratio": OR,
    "CI Lower": CI_lower,
    "CI Upper": CI_upper,
    "p-value": result.pvalues
})

print(summary_table)

Looking at p-values (<0.05)

| Variable          | Odds Ratio         | p-value | Interpretation                |
| ----------------- | ------------------ | ------- | ----------------------------- |
| cd4_count         | 1.003              | 0.082   | Not statistically significant |
| Occupation groups | extremely small OR | >0.98   | Not significant               |
| Current regimen   | OR < 1             | >0.11   | Not significant               |


No variable remained significantly associated with depression.